File: alogrithm/sort/select/ssort.c
#include<stdio.h>
#include<stdlib.h>

void ssort(int array[],int n)
{
	int i,j;
	int small;
	for(i=0;i<n;i++)
	{
		small=i;
		printf("small=%d\n ",small);
		for(j=i+1;j<n;j++)
		{
			if (array[j]<array[small])
				small=j;
		}
		int temp=0;
		if (small!=i)
		{
			temp=array[i];
			array[i]=array[small];
			array[small]=temp;
			printf("array[%d]=%d,array[small]=%d\n ",i,array[i],array[small]);
		}
	}
	for(i=0;i<n;i++)
		printf("%d ",array[i]);

}


int main(void)
{
	int array[6]={7,3,1,6,4,8};
	ssort(array,6);
}

File: alogrithm/sort/select/ssort.py

def ssort(array):

    for i in xrange(0,len(array)):
        j=i+1
        small=i
        while j<len(array):
            if array[j]<array[small]:
                small=j;
            j+=1
        if i!=small:
            array[small],array[i] = array[i],array[small]

    return array




array=[2,62,7,3,8,1,1]
print 'any',all(array)
print ssort(array)

